save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/internal/UsersController.kt

```diff
@@ -44,11 +47,12 @@ class UsersController(
     @PostMapping("/new")
     fun saveNewUser(@RequestBody user: User) {
         val userName = requireNotNull(user.name) { "Provided user $user doesn't have a name" }
-        userRepository.findByName(userName)?.let {
+        originalLoginRepository.findByName(userName)?.user?.let {
             logger.debug("User $userName is already present in the DB")
         } ?: run {
             logger.info("Saving user $userName to the DB")
-            userRepository.save(user)
+            val newUser = userRepository.save(user)
+            originalLoginRepository.save(OriginalLogin(user.name, newUser, user.source))
```

Comment:
This should be encapsulated in `UserDetailsService`

Reply:
Fixed